src/main.rs

```diff
@@ -0,0 +1,160 @@
+        if utils::is_python_project(&context.app_dir).map_err(BuildpackError::DetectIo)? {
+            DetectResultBuilder::pass().build()
+        } else {
+            log_info("No Python project files found (such as requirements.txt).");
+            DetectResultBuilder::fail().build()
+        }
+    }
+
+    fn build(&self, context: BuildContext<Self>) -> libcnb::Result<BuildResult, Self::Error> {
+        // We perform all project analysis up front, so the build can fail early if the config is invalid.
+        // TODO: Add a "Build config" header and list all config in one place?
+        let is_function = salesforce_functions::is_function_project(&context.app_dir)
+            .map_err(BuildpackError::ProjectDescriptor)?;
+        let package_manager = package_manager::determine_package_manager(&context.app_dir)
+            .map_err(BuildpackError::DeterminePackageManager)?;
+
+        log_header("Determining Python version");
+        let python_version = python_version::determine_python_version(&context.app_dir)
+            .map_err(BuildpackError::PythonVersion)?;
+        let packaging_tool_versions = PackagingToolVersions::default();
+
+        // We inherit the current process's env vars, since we want `PATH` and `HOME` to be set
+        // so that later commands can find tools like Git in the stack image. Any user-provided
+        // env vars will still be excluded, due to the use of `clear-env` in `buildpack.toml`.
+        let mut command_env = Env::from_current();
+
+        // Create the layer containing the Python runtime, and the packages `pip`, `setuptools` and `wheel`.
+        log_header("Installing Python and packaging tools");
+        let python_layer = context.handle_layer(
+            layer_name!("python"),
+            PythonLayer {
+                command_env: &command_env,
+                python_version: &python_version,
+                packaging_tool_versions: &packaging_tool_versions,
+            },
+        )?;
+        command_env = python_layer.env.apply(Scope::Build, &command_env);
+
+        // Create the layers for the application dependencies and package manager cache.
+        // In the future support will be added for package managers other than pip.
+        let dependencies_layer_env = match package_manager {
+            PackageManager::Pip => {
+                log_header("Installing dependencies using Pip");
+                let pip_cache_layer = context.handle_layer(
+                    layer_name!("pip-cache"),
+                    PipCacheLayer {
+                        python_version: &python_version,
+                        packaging_tool_versions: &packaging_tool_versions,
+                    },
+                )?;
+                let pip_layer = context.handle_layer(
+                    layer_name!("dependencies"),
+                    PipDependenciesLayer {
+                        command_env: &command_env,
```

Comment:
You're passing in the current environment. I think you also need to apply `context.platform` environment (assuming that users will want to do things like use env vars to do things, like setting private resource creds).

Reply:
Sounds intentional 👍 

Wording wise "due to the use of `clear-env` in `buildpack.toml`. " is not terribly informative for me. Looking up the docs:

```
When clear-env in buildpack.toml is set to true for a given buildpack, the lifecycle MUST NOT set user-provided environment variables in the environment of /bin/detect or /bin/build.
```

I think that `clear-env` info is an implementation detail of the exclusion. I would say the more important reason is WHY they're being excluded. Maybe something like:

```
All user-provided env vars are intentionally excluded to prevent users from accidentally setting an unexpected configuration value. This behavior matches the classic buildpack.
```

Reply:
192230568b35c7ac32e20677e4d11838dc8b0659